lib/kubernetes-deploy/kubectl.rb

```diff
@@ -20,26 +21,35 @@ def initialize(namespace:, context:, logger:, log_failure_by_default:, default_t
       raise ArgumentError, "context is required" if context.blank?
     end
 
-    def run(*args, log_failure: nil, use_context: true, use_namespace: true, raise_if_not_found: false)
+    def run(*args, log_failure: nil, use_context: true, use_namespace: true, raise_if_not_found: false, attempts: 1)
       log_failure = @log_failure_by_default if log_failure.nil?
 
       args = args.unshift("kubectl")
       args.push("--namespace=#{@namespace}") if use_namespace
       args.push("--context=#{@context}")     if use_context
       args.push("--request-timeout=#{@default_timeout}") if @default_timeout
-
-      @logger.debug Shellwords.join(args)
-      out, err, st = Open3.capture3(*args)
-      @logger.debug(out.shellescape) unless output_is_sensitive?
-
-      unless st.success?
-        if log_failure
-          @logger.warn("The following command failed: #{Shellwords.join(args)}")
-          @logger.warn(err) unless output_is_sensitive?
-        end
-
-        if raise_if_not_found && err.match(NOT_FOUND_ERROR_TEXT)
-          raise ResourceNotFoundError, err
+      out, err, st = nil
+
+      (1..attempts).to_a.each do |attempt|
```

Comment:
`attempts.times`?

Reply:
That would yield 0 to 4 instead of 1 to 5, so there'd be a tradeoff in less elegant code below where `attempt` is used.